Setup configuration-aws-icp

hasheddan opened the pull request:
<!--
Thank you for helping to improve Upbound!

Please read through https://git.io/fj2m9 if this is your first time opening an
Upbound pull request. Find us in https://slack.crossplane.io/messages/upbound if
you need any help contributing.
-->

### Description of your changes

<!--
Briefly describe what this pull request does. Be sure to direct your reviewers'
attention to anything that needs special consideration.

We love pull requests that resolve an open issue. If yours does, use the below
line to indicate which issue your PR fixes, for example "Fixes #500":
-->

Sets up APIs and dependencies for the AWS internal cloud platform package.

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change. Consider pasting snippets
with the commands or configurations you used to test, in case you or a reviewer
needs to repeat the test in future.
-->

e2e testing in progress.

hasheddan committed:
Add crossplane.yaml with dependencies

Adds a crossplane.yaml with dependencies on other configuration
packages.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

hasheddan committed:
Add virtual machine API

Adds an API for a virtual machine that maps to an EC2 instance.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

hasheddan committed:
Add a VM example

Adds an example manifest for provisioning a VM.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

cwilhit commented:
There's one bug in the VM composition that needs to be fixed before this goes live, I think. The AMI # changes for these images depending on if you've chosen us-west or us-east.

hasheddan committed:
Use nested composition to select proper region implementation

Adds an intermediate XRD to allow for mapping to different compositions
based on region supplied to outer XR.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

hasheddan reviewed (commented):
@cwilhit still in process of testing, but updated to use nested composition to address this issue.

hasheddan reviewed (commented):
Verified we are now working correctly!

```
🤖 (configuration-aws-icp) k get instance.ec2 
NAME                READY   SYNCED   EXTERNAL-NAME         AGE
my-vm-28smr-qlfrq   True    True     i-01a4660357698dd8e   8m17s
🤖 (configuration-aws-icp) k get claim
NAME    SYNCED   READY   CONNECTION-SECRET   AGE
my-vm   True     True                        27m
🤖 (configuration-aws-icp) k get composite
NAME                                                        SYNCED   READY   COMPOSITION                                    AGE
xvirtualmachine.aws.compute.starter.org/my-vm-28smr-p2z69   True     True    east.virtualmachines.aws.compute.starter.org   24m

NAME                                              SYNCED   READY   COMPOSITION                               AGE
xvirtualmachine.compute.starter.org/my-vm-28smr   True     True    virtualmachines.aws.compute.starter.org   27m
```
